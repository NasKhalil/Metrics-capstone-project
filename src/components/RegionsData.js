import React from 'react';
import PropTypes from 'prop-types';

import '../styling/RegionsData.css';

const RegionsData = ({ regions, countryName }) => (
  <>
    <div className="region">
      <h3 className="region-title">Region&apos;s State</h3>
      <ul className="regions">
        {regions && regions.map((region) => {
          const {
            name,
            today_confirmed: todayConfirmed,
            today_deaths: todayDeaths,
            today_recovered: todayRecovered,
          } = region;
          return (
            <li key={name} className="region-list">
              <div>
                <h3 className="region-name">{name}</h3>
                <div className="region-detail">
                  <div>
                    <span>Total Confirmed </span>
                    <h3>{`${todayConfirmed} cases`}</h3>
                  </div>
                  <div>
                    <span>Total Deaths</span>
                    <h3>{`${todayDeaths} cases`}</h3>
                  </div>
                  <div>
                    <span>Total Recovered</span>
                    <h3>{`${todayRecovered} cases`}</h3>
                  </div>
                </div>
              </div>
            </li>
          );
        })}
      </ul>

      {(!regions || !regions[0]) && (
        <h3 className="noData">
          We are sorry!
          <br />
          Data will be available as soon as possible for
          {' '}
          {countryName}
        </h3>
      )}
    </div>
  </>
);

RegionsData.propTypes = {
  regions: PropTypes.instanceOf(Array).isRequired,
  countryName: PropTypes.string.isRequired,
};

export default RegionsData;
