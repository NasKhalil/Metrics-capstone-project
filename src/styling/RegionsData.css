.region {
  color: #fff;
  border: 1px solid;
  border-radius: 8px;
  padding: 24px;
  width: 60%;
  background-color: #5f5b5b;
  opacity: 0.6;
  margin: 10px auto;
}

.region-title {
  margin: 12px;
  font-size: 28px;
}

@media all and (min-width: 1024px) {
  .region {
    color: #fff;
    border: 1px solid;
    border-radius: 8px;
    padding: 24px;
    width: 60%;
    background-color: #5f5b5b;
    opacity: 0.6;
    margin: 10px auto;
  }

  .region-title {
    margin: 16px;
    font-size: 32px;
  }
}
