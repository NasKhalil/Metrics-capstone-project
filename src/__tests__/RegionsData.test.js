import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RegionsData from '../components/RegionsData';

describe('RegionsData test', () => {
  test('should render', () => {
    const footer = render(
      <Router>
        <Provider store={store}>
          <RegionsData regions={[]} countryName="France" />
        </Provider>
      </Router>,
    );

    expect(footer).toMatchSnapshot();
  });
});
