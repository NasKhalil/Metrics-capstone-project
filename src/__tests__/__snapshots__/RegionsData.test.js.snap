// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionsData test should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="region"
      >
        <h3
          class="region-title"
        >
          Region's State
        </h3>
        <ul
          class="regions"
        />
        <h3
          class="noData"
        >
          We are sorry!
          <br />
          Data will be available as soon as possible for
           
          France
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="region"
    >
      <h3
        class="region-title"
      >
        Region's State
      </h3>
      <ul
        class="regions"
      />
      <h3
        class="noData"
      >
        We are sorry!
        <br />
        Data will be available as soon as possible for
         
        France
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
